src/CoreServices/FSEvents.cs

```diff
@@ -154,9 +154,11 @@ public void Dispose ()
 
 		protected virtual void Dispose (bool disposing)
 		{
-			if (handle != IntPtr.Zero) {
-				FSEventStreamRelease (handle);
-				handle = IntPtr.Zero;
+			if (disposing) {
```

Comment:
Why only when disposing? We'll never release the unmanaged handle otherwise?

Reply:
This follows the disposing pattern - https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose

Reply:
Yes, but in this case, an IntPtr does not have a finalizer.

[SafeHandle-derived](https://source.dot.net/#System.Private.CoreLib/SafeHandle.cs,69) classes have a finalizer run by default, so what that means is:
a) your class is finalized, then safehandle is finalized
b) your class is disposed, then your safehandle is disposed

With the current code, it's just a plain IntPtr. What it means for the finalizer case is that when `FSEventStream` is finalized, _nothing happens_. Which means that the native handle is never released, so every FSEventStream instance that is not explicitly disposed _will leak_.

Reply:
See the pattern which doesn't use a safe-handle, specifically:
```csharp
   // Protected implementation of Dispose pattern.
   protected virtual void Dispose(bool disposing)
   {
      if (disposed)
         return; 
      
      if (disposing) {
         // Free any other managed objects here.
         //
      }
      
      // Free any unmanaged objects here.
      // <---- FSEventStreamRelease here
      disposed = true;
   }

```